src/parsers/manifest/dash/node_parsers/Period.ts

```diff
@@ -43,6 +67,508 @@ export interface IPeriodAttributes {
   start? : number;
   duration? : number;
   bitstreamSwitching? : boolean;
+  linkURL? : string;
+  resolveAtLoad? : boolean;
+}
+
+/**
+ * Create an intermediate period object from Element.
+ * @param {Element} root
+ * @returns {Object}
+ */
+function createIntermediatePeriod(root: Element) {
+  function parseRootChildren(rootChildren : NodeList) {
+    const periods : IPeriodIntermediateRepresentation[] = [];
+    for (let i = 0; i < rootChildren.length; i++) {
+      if (rootChildren[i].nodeType === Node.ELEMENT_NODE) {
+        const currentNode = rootChildren[i] as Element;
+        switch (currentNode.nodeName) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this "switch" statement with "if" statements to increase readability. ([typescript:S1301](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS1301))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=337&issues=AWZoys-TP8lmccBDy_QJ&open=AWZoys-TP8lmccBDy_QJ)

[ISSUE_KEY]: AWZoys-TP8lmccBDy_QJ